### PR TITLE
Enable webrtc by default, enable webrtc wpt tests

### PR DESCRIPTION
**Pull request by Manishearth**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23377)

### CODE REVIEW

File: tests/wpt/include.ini [hunk 1]
```diff
@@ -129,6 +129,8 @@ skip: true
   skip: false
 [webvr]
   skip: false
+[webvr]
```
Comment (jdm) [inline tests/wpt/include.ini]: Wrong directory.

**Comment by Manishearth** [comment]
Alright, looks like we're missing a webrtcbin element. I guess this needs to be explicitly enabled?

**Comment by Manishearth** [comment]
Oh, right, mach boostrap just fixes it for users.

https://github.com/servo/saltfs/pull/960

**Comment by Manishearth** [comment]
The mediacapture tests will fail because we have no media devices :|

Not sure what we can do here, perhaps I should just remove that commit.

**Comment by Manishearth** [comment]
So I've included the media capture test results so we can continue to hack on this locally (there are some failing tests I'm interested in and I'd like to have a measure of progress there). However I've removed it from include.ini, until we can figure out how to emulate capture devices.

**Comment by Manishearth** [comment]
Hmm, same failure :/

**Comment by Manishearth** [comment]
The local chroot build definitely had webrtc installed. I'll check what the server says in a bit

**Comment by jdm** [comment]
`./gst/lib/libgstwebrtc-1.0.so.0` was present in the files that were unzipped.

**Comment by Manishearth** [comment]
I think the scanner looks in the `lib/x86_64-linux-gnu/gstreamer-1.0` folder. On my system there's a libgstwebrtc.so there (in addition to `lib/x86_64-linux-gnu/libgstwebrtc-1.0.so` -- the files are different), but this tarball doesn't have this.

OTOH on the chrooted system where I built this, gst-inspect-1.0 works fine, so there's something I'm missing in the packaging steps.

**Comment by Manishearth** [comment]
The files are definitely there in the tarball

**Comment by Manishearth** [comment]
Oh, wrong folder, we should be looking at `gst/`, the `gstreamer` folder is old.

The bin folder doesn't have execute permissions, it seems.

**Comment by Manishearth** [comment]
Even with exec permissions fixed it still doesn't work

**Comment by Manishearth** [comment]
I think we first need to set `overwrite=True` in the archive.extracted state, but that's not all of it.

**Comment by Manishearth** [comment]
Okay, so there are multiple issues here:

 - The [gst salt stuff](https://github.com/servo/saltfs/blob/master/servo-build-dependencies/linux-gstreamer.sls) gets rid of execute perms on a bunch of stuff. It shouldn't. 
 - My Trusty build relies on libgnutls-26, Xenial (which our build machines are on) has libgnutls-30. This matters, apparently. I could potentially build and bundle libgnutls along with gst, but I'm not sure if I should. Another solution is to make separate packages for trusty and xenial.

**Comment by Manishearth** [comment]
https://bugzilla.gnome.org/show_bug.cgi?id=794864

**Comment by Manishearth** [comment]
Looks like this is all fixed in 1.15.

Maybe I can just package up 1.15.1 and see what happens?

**Comment by Darkspirit** [comment]
Should this be updated as well?
https://github.com/servo/servo/blob/ed5c1cd6c24b56e47808b76a9197a5c030003642/.travis.yml#L26

**Comment by Manishearth** [comment]
Doesn't quite matter, we don't run anything on travis yet. But I'll do it before merging.

**Comment by Manishearth** [comment]
What i'll probably end up doing is landing the gstreamer upgrades in a separate PR, but I'm doing the testing here since this lets us know if the webrtc stuff works

**Comment by Manishearth** [comment]
```
  │ 0:00:00.368563863 75845 0x7fd5fc1a8850 ERROR              webrtcbin gstwebrtcbin.c:3540:_set_description_task:<sendrecv> media 1 has different ice-ufrag values in bundle. ETEn != BGKk
  │ 0:00:00.373503936 75845 0x7fd5fc1a8850 ERROR              webrtcbin gstwebrtcbin.c:2402:_create_answer_task:<sendrecv> Asked to create an answer without a remote description
```

this is new

**Comment by Manishearth** [comment]
Actually a lot of these issues might be due to the rebase, since I have lots of "fail expected notrun" and "ok expected error"s too

**Comment by Manishearth** [comment]
Or the gstreamer upgrade :)

**Comment by Manishearth** [comment]
Even more reason to split out the gstup into a separate PR

**Comment by Manishearth** [comment]
Looks like we also need `vp8enc` to work

**Comment by Manishearth** [comment]
We need libvpx 1 (not just libvpx). Fun.

**Comment by Manishearth** [comment]
Wait, actually, we should be generating these binaries on xenial schroots, totally forgot to try this.

The problem is that xenial doesn't have libvpx1 (it has libvpx3). There's no need for me to try and make this work with libvpx1 anyway.

**Comment by Manishearth** [comment]
@bors-servo retry

deployed a thing with VP8 working

**Comment by Manishearth** [comment]
Hmm, that test was crashing on CI (but not locally) earlier as well. Unsure what's going on, maybe I should just exclude it.

(Maybe its a difference between 1.14 and 1.16?)

**Comment by ferjm** [comment]
```
 ▶ CRASH [expected OK] /webrtc/protocol/ice-state.https.html
  │ 
  │ VMware, Inc.
  │ softpipe
  │ 3.3 (Core Profile) Mesa 18.3.0-devel
  │ 
  │ (servo:104162): GStreamer-WARNING **: External plugin loader failed. This most likely means that the plugin loader helper binary was not found or could not be run. You might need to set the GST_PLUGIN_SCANNER environment variable if your setup is unusual. This should normally not be required though.
  │ 0:00:00.023881834 104162 0x55bb1a44e530 WARN      GST_PLUGIN_LOADING gstplugin.c:793:_priv_gst_plugin_load_file_for_registry: module_open failed: libmms.so.0: cannot open shared object file: No such file or directory
  │ 
  │ (servo:104162): GStreamer-WARNING **: Failed to load plugin '/home/servo/gst/lib/gstreamer-1.0/libgstmms.so': libmms.so.0: cannot open shared object file: No such file or directory
  │ 0:00:00.024117241 104162 0x55bb1a44e530 WARN      GST_PLUGIN_LOADING gstplugin.c:793:_priv_gst_plugin_load_file_for_registry: module_open failed: libfaad.so.2: cannot open shared object file: No such file or directory
  │ 
  │ (servo:104162): GStreamer-WARNING **: Failed to load plugin '/home/servo/gst/lib/gstreamer-1.0/libgstfaad.so': libfaad.so.2: cannot open shared object file: No such file or directory
  │ 0:00:00.334256857 104162 0x7f53e41a9ed0 ERROR              webrtcbin 
```

It seems that we are missing a couple of plugins dependencies: `libmms` and `libfaad`

**Comment by jdm** [comment]
To be fair, we have always been missing those and those messages appear in the output of any test that loads gstreamer.

**Comment by Manishearth** [comment]
Yeah. Note that the gst bundle instructions do list those packages at the end. But I don't think they're causing these problems.

**Comment by Manishearth** [comment]
Ah, I was running 1.14 locally, and this crash is new.

@bors-servo try

**Comment by Manishearth** [comment]
@bors-servo try

now i have failures i see locally but not on CI. whee.

**Comment by Manishearth** [comment]
libgstgl-1.0.so.0 is failing now

**Comment by Manishearth** [comment]
That file is on the server, I checked :/

**Comment by Manishearth** [comment]
This should pass, but It's unclear to me why there's a discrepancy between this and what runs locally

**Comment by Manishearth** [comment]
Closing for now.

**Pull request closed without merging**